File: android/src/main/kotlin/com/polyfence/core/PolyfenceCoreDelegate.kt
package com.polyfence.core

/**
 * Interface for platform bridges (Flutter, React Native, etc.) to receive
 * events from the core engine. Each bridge implements this interface.
 */
interface PolyfenceCoreDelegate {
    /** Called when a geofence event occurs (ENTER, EXIT, DWELL, RECOVERY_*) */
    fun onGeofenceEvent(eventData: Map<String, Any>)

    /** Called on each GPS location update */
    fun onLocationUpdate(locationData: Map<String, Any>)

    /** Called with runtime performance/health status */
    fun onPerformanceEvent(performanceData: Map<String, Any>)

    /** Called when an error occurs */
    fun onError(errorData: Map<String, Any>)
}

File: ios/Classes/PolyfenceCoreDelegate.swift
import Foundation
import CoreLocation

/// Protocol for platform bridges (Flutter, React Native, etc.) to receive
/// events from the core engine. Each bridge implements this protocol.
public protocol PolyfenceCoreDelegate: AnyObject {
    /// Called when a geofence event occurs (ENTER, EXIT, DWELL, RECOVERY_*)
    func onGeofenceEvent(_ eventData: [String: Any])

    /// Called on each GPS location update
    func onLocationUpdate(_ locationData: [String: Any])

    /// Called with runtime performance/health status
    func onPerformanceEvent(_ performanceData: [String: Any])

    /// Called when an error occurs
    func onError(_ errorData: [String: Any])
}
